lib/utils/createConfig.js

```diff
@@ -14,12 +14,12 @@ function createConfig(config, argv, { port }) {
     options.bonjour = true;
   }
 
-  if (argv.host !== 'localhost' || !options.host) {
+  if (argv.host && (argv.host !== 'localhost' || !options.host)) {
```

Comment:
Set host only when `CLI` arg specify

Reply:
Same logic for other arguments (otherwise it is `undefined` and it makes no sense